sensing/gnss_poser/src/gnss_poser_core.cpp

```diff
@@ -135,18 +139,22 @@ void GNSSPoser::callbackNavSatFix(
     canGetCovariance(*nav_sat_fix_msg_ptr) ? nav_sat_fix_msg_ptr->position_covariance[4] : 10.0;
   gnss_base_pose_cov_msg.pose.covariance[6 * 2 + 2] =
     canGetCovariance(*nav_sat_fix_msg_ptr) ? nav_sat_fix_msg_ptr->position_covariance[8] : 10.0;
-  gnss_base_pose_cov_msg.pose.covariance[6 * 3 + 3] = 0.1;
-  gnss_base_pose_cov_msg.pose.covariance[6 * 4 + 4] = 0.1;
-  gnss_base_pose_cov_msg.pose.covariance[6 * 5 + 5] = 1.0;
+  gnss_base_pose_cov_msg.pose.covariance[6 * 3 + 3] =
+    orientation_msg_ != nullptr ? std::pow(orientation_msg_->orientation.rmse_rotation_x, 2) : 0.1;
+  gnss_base_pose_cov_msg.pose.covariance[6 * 4 + 4] =
+    orientation_msg_ != nullptr ? std::pow(orientation_msg_->orientation.rmse_rotation_y, 2) : 0.1;
+  gnss_base_pose_cov_msg.pose.covariance[6 * 5 + 5] =
+    orientation_msg_ != nullptr ? std::pow(orientation_msg_->orientation.rmse_rotation_z, 2) : 1.0;
   pose_cov_pub_->publish(gnss_base_pose_cov_msg);
 
   // broadcast map to gnss_base_link
   publishTF(map_frame_, gnss_base_frame_, gnss_base_pose_msg);
 }
 
-void GNSSPoser::callbackNavPVT(const ublox_msgs::msg::NavPVT::ConstSharedPtr nav_pvt_msg_ptr)
+void GNSSPoser::callbackAutowareOrientation(
+  const autoware_sensing_msgs::msg::GnssInsOrientationStamped msg)
 {
-  nav_pvt_msg_ptr_ = nav_pvt_msg_ptr;
+  orientation_msg_->orientation = msg.orientation;
```

Comment:
```suggestion
  orientation_msg_ = std::make_shared<autoware_sensing_msgs::msg::GnssInsOrientationStamped>(msg);
```

if type of `msg` changes `autoware_sensing_msgs::msg::GnssInsOrientationStamped::ConstSharedPtr`, 
```suggestion
  orientation_msg_ = msg;
```

Reply:
> if you change type of `msg`to `autoware_sensing_msgs::msg::GnssInsOrientationStamped::ConstSharedPtr`,

Hello. Thanks for your quick review. I changed type of `msg`to `autoware_sensing_msgs::msg::GnssInsOrientationStamped::ConstSharedPtr`. But `orientation_msg_` is SharedPtr and msg type is `ConstSharedPtr`. When I changed it as in your suggestion I got an error so I changed this line like here

`orientation_msg_->orientation = msg->orientation;`

Reply:
I cant change the type of `orientation_msg_` to `ConstSharedPtr` because I want assign values to orientation msg. Therefore it must be not `Const`. In addition I want to use pointer for `orientation_msg_` so I cant change type of `orientation_msg_`. 
However you are right there is nothing in `orientation_msg_->header`. So I add another line for fill the `orientation_msg_->header`.

Reply:
I see, OK!
Fixed! https://github.com/autowarefoundation/autoware.universe/pull/1230/commits/77751fbdc49abb44b3c36f140e2c1caf9baa68f7